### ISSUE TITLE
[Bug]: grf vehicle without random reverse CB inherits flip when autoreplacing a vehicle that has random reverse CB and is flipped

### ISSUE THREAD

**Issue by andythenorth** [OpenTTD/OpenTTD#12100]
### Version of OpenTTD

Post https://github.com/OpenTTD/OpenTTD/pull/11489

### Expected result

CBID_VEHICLE_BUILD_PROBABILITY CB 162 was a useful addition in https://github.com/OpenTTD/OpenTTD/pull/11489 

However when auto-replacing a flipped vehicle, the replacement vehicle inherits the flag. This is traditional behaviour, not arising directly from CB 162, but is more easily triggered with CB 162 in use.

I don't know what expected behaviour we want but the current result can be not ideal.

### Actual result

Intended (engine uses CB 162 at 50:50 ratio)
<img width="231" alt="image" src="https://github.com/OpenTTD/OpenTTD/assets/1780327/5b347786-d519-43f6-bd66-f3ab304f5a27">

Unintended - some of the leading cab cars are reversed (yellow is at wrong end).

<img width="261" alt="image" src="https://github.com/OpenTTD/OpenTTD/assets/1780327/5e6f32da-8e6d-4b77-b368-35627c22820e">

**Possible solutions**

- in grf, for vehicles that shouldn't be flipped, use CB 162 to force 100% chance of not being flipped?  (will this invert the current flip bit from autoreplace? - haven't tested)
- in grf, disallow flip (using flag) for vehicles that could be affected by this?
- in grf, detect the flip state, and reverse the sprites?
- reset the flip bit on autoreplace? Always?  Or on a flag?  Or if CB 162 isn't handled?

### Steps to reproduce

- Using Iron Horse 3.6.0
- Set game date to around 2000.
- Turn off vehicle expiry.
- Build Larks + Autocoaches, enough that some Larks randomly flip on build.
- Set autoreplace to 
  - Lark > Clipper (single)
  - Autocoach > Clipper (double)
- Force autoreplace in depot.

**Comment by PeterN** [comment]
Same effect with cloning a train with the reversed bit set. If the source vehicles are flipped they will be flipped (again) in the new train, and the random chance will happen, so if they are not flipped originally, they could still be flipped after cloning/autoreplacing.

Using the callback to force 100% chance of not flipping won't help as that occurs before the flipped bit is actioned.

This boils down to:

1) What is the expected behaviour of renewing/autoreplacing/cloning a manually flipped vehicle.
2) What is the expected behaviour of renewing/autoreplacing/cloning a randomly flipped vehicle.

The code that copies the existing flipped state was designed for when there was only manual flipping, so doesn't take into account random flipping. If the source is flipped, then the new vehicle will be flipped again, potentially unflipping it...

**Comment by PikkaBird** [comment]
> This boils down to

I don't think it makes much sense to treat manually vs randomly flipped vehicles differently; flipped is flipped, and what matters is the appropriateness of flipping the *new* vehicle, not the old one.

For cloning and autorenew, I don't think there's a problem; they're the same vehicle, so the same flipping appropriateness applies.

For autoreplace, the issue is one that exists generally due to the redefinition of the prop 27 bit 3 in #10262 - single vehicles with a defined "front" are now flippable, with no way for grf authors to prevent it. If the GRF author wants to prevent these vehicles being graphically reversed - due to either manual or inherited flipping - there are two solutions I can see, neither of which need any addition to the GRF spec or changes to the game:

1) code the vehicle as articulated.
2) provide explicit reversed-flipped sprites, so the vehicle looks the same whether it's reversed or not.

The third option, which would require changes to the game, is to add a new flag which replicates the old function of prop 27 bit 3, but that would still require updating NewGRFs to add the flag.

**Comment by PikkaBird** [comment]
... But then of course, we run into another problem; with vehicles like the Clipper (or the single 101s in Sunshine), you might want it flipped at the *back* of the train, but not the front...

Maybe it's just a case of BAD FEATURES, and there's no one-size-fits-all solution here.
